tests/test_script.py

```diff
@@ -143,3 +169,23 @@ def test_full_args(self):
         name="progname"
         assert output == FULL_OUTPUT.format(name=name, NAME=name.upper(),
                                             DATE=DATE)
+
+    def test_pyproject_toml(self):
+        """
+        Test that we can read information from pyproject.toml.
+        """
+        current_dir = os.getcwd()
+        try:
+            os.chdir(self.workdir)
+            with open("pyproject.toml", "w+") as script_fd:
+                script_fd.write(PYPROJECT_TOML_FILE_CONTENTS.format(gitdir=current_dir))
+            with open("setup.py", "w+") as script_fd:
+                script_fd.write(SETUP_PY_FILE_CONTENTS)
+
+            os.mkdir("somefile")
+            with open(os.path.join("somefile", "__init__.py"), "w+") as script_fd:
+                script_fd.write(SIMPLE_FILE_CONTENTS.format(ap_arguments=""))
+
+            assert 0 == run_setup_py(["build"])
```

Comment:
We still should check that the built output matches our expectations, otherwise LGTM.

Reply:
Done, good point.